CASMPET-6934 iSCSI SBPS: md doc for worker node(s) personalization (#5357)
Cray-HPE/docs-csm: operations/configuration_management/iSCSI_SBPS_Node_Personalization.md

# iSCSI SBPS worker node(s) personalization

As part of the iSCSI SBPS solution, we need to setup/ configure worker nodes as iSCSI targets (servers) with necessary provisioning, configuration and enablement of required components.
In order to achieve this, CFS based Ansible plays (`csm-config`) are created and deployed in order to

* provision `iSCSI targets`/`LIO services` and present the `LIO network` service via the HSN and NMN IP networks (TCP port 3260 by default), via iSCSI Portals.
  
* enable SBPS Marshal Agent by installing the agent rpm and starting the respective `systemd` service (`sbps-marshal`).
  
* enable `spire` for authentication used for `IMS/ S3` access.
  
* create `DNS SRV and A records` in order to discover iSCSI targets from iSCSI initiators/clients (passed as boot parameter).
  
* mount S3 images (`boot-images` bucket) with new, dedicated S3 user (`ISCSI-SBPS`) read only bucket policy via `s3fs`.

>   

Reviewer: The blank lines have trailing whitespace, please remove them.

Author: Fixed.

cfs-b840455c-e919-4656-b64a-44d433f082dc-5d4jp
```

```text
ncn-m001:/home/csm-config-management #  kubectl logs -n services  cfs-b840455c-e919-4656-b64a-44d433f082dc-5d4jp ansible
Waiting for Inventory
Inventory generation completed
SSH keys migrated to /root/.ssh
...
Sidecar available
Running ansible/config_sbps_iscsi_targets.yml from repo https://api-gw-service-nmn.local/vcs/cray/csm-config-management.git
....
All playbooks completed successfully
```

### Important things to check after worker node(s) personalization

#### Check to see if the SBPS Marshal Agent is running without any errors

**E.g.**

```text
ncn-w001:/home # systemctl status sbps-marshal
● sbps-marshal.service - System service that manages Squashfs images projected via iSCSI for IMS, PE, and other ancillary images similar to PE.
     Loaded: loaded (/usr/lib/systemd/system/sbps-marshal.service; enabled; vendor preset: disabled)
     Active: active (running) since Thu 2024-08-22 11:57:48 UTC; 2 weeks 4 days ago
   Main PID: 2878373 (sbps-marshal)
      Tasks: 1
     CGroup: /system.slice/sbps-marshal.service
             └─ 2878373 /usr/lib/sbps-marshal/bin/python /usr/lib/sbps-marshal/bin/sbps-marshal
...
```

> #### Check to see if the images/ LUN mapping created under `targetcli ls` cmd

Reviewer 2: ```suggestion
#### Check to see if the images/ LUN mapping are created with the `targetcli ls` command
```

Author: Fixed.

# iSCSI SBPS worker node(s) personalization

> 

Reviewer: This whole file has `\r\n` line terminations, they should be just `\n`.  Please strip out all the `\r`s from this file.

> 

Reviewer: You can't see it, but this whole file has `\r\n` line terminations, they should be just `\n`.  Please strip out all the `\r`s from this file.

Author: Fixed

As part of the iSCSI SBPS solution, we need to setup/ configure worker nodes as iSCSI targets (servers) with necessary provisioning, configuration and enablement of required components.
In order to achieve this, CFS based Ansible plays (csm-config) are created and deployed in order to
* provision `iSCSI targets`/ `LIO services` and present the `LIO network` service via the HSN and NMN IP networks (TCP port 3260 by default), via iSCSI Portals.
* enable SBPS Marshal Agent by installing the agent rpm and starting the respective systemd service (`sbps-marshal`)
* enable `spire` for authentication used for `IMS/ S3` access
* create `DNS SRV and A records` in order to discover iSCSI targets from iSCSI initiators/clients (passed as boot parameter)
* mount s3 images (`boot-images` bucket) with new, dedicated s3 user (`ISCSI-SBPS`) read only bucket policy via s3fs
* apply k8s label (`iscsi=sbps`), to be consumed by the Goss test suit and LIO Metrics for identification of nodes on which these have to run.

> Target worker node selection is via dynamic inventory stored in HSM (Hardware State Manager) – the default is all worker nodes (`Management_Worker` group).

Reviewer: Not easy to see, but in `Manager) – the` the dash is not a simple ascii dash character, please change it to `Manager) - the`

Author: Fixed

## Prerequisites

* Respective csm-config version CFS Ansible plays are available at VCS (Version Control Service)
* On cluster nodes, the VCS can be accessed through the gateway. VCS credentials for the `crayvcs` user are required before cloning a repository.
  See [VCS Administrative User](Version_Control_Service_VCS.md#vcs-administrative-user) in
  [Version Control Service (VCS)](Version_Control_Service_VCS.md) for more information.

Below is the procedure (with examples) that need to be followed for worker node personalization for iSCSI SBPS. Please note that the "Optional" steps mentioned below
are only to get the latest version of the `csm-config` or to check if the known/ intended version of the `csm-config` is available before proceeding with node personalization
with creating the CFS config/ CFS session.

## Procedure

### **Check the VCS availability**

```
ncn-m001:/home # /usr/local/bin/cmsdev test -q vcs
Starting main run, version: 1.22.0, tag: hMUDa
Starting sub-run, tag: hMUDa-vcs
Ended sub-run, tag: hMUDa-vcs (duration: 3.982640351s)
Ended run, tag: hMUDa (duration: 3.994254571s)
SUCCESS: All 1 service tests passed: vcs

> ncn-m001:/home #

Reviewer 2: ```suggestion

```
This is just my opinion and it works as is, but I would remove the prompt at the end of each block. I do not think you need to leave the `ncn-m001:/home #` at the end of the output. If you choose to remove it, be sure to remove the ending prompts in the rest of the document.

Author: Fixed

Waiting for Inventory
Inventory generation completed
SSH keys migrated to /root/.ssh
...
Sidecar available
Running ansible/config_sbps_iscsi_targets.yml from repo https://api-gw-service-nmn.local/vcs/cray/csm-config-management.git
....
All playbooks completed successfully
ncn-m001:/home/csm-config-management #
```
### **Important things to check after worker node(s) personalization**

#### **Check to see if the SBPS Marshal Agent is running without any errors.**

**E.g.**

```
ncn-w001:/home # systemctl status sbps-marshal
● sbps-marshal.service - System service that manages Squashfs images projected via iSCSI for IMS, PE, and other ancillary images similar to PE.
     Loaded: loaded (/usr/lib/systemd/system/sbps-marshal.service; enabled; vendor preset: disabled)
     Active: active (running) since Thu 2024-08-22 11:57:48 UTC; 2 weeks 4 days ago
   Main PID: 2878373 (sbps-marshal)
      Tasks: 1
     CGroup: /system.slice/sbps-marshal.service
             └─ 2878373 /usr/lib/sbps-marshal/bin/python /usr/lib/sbps-marshal/bin/sbps-marshal
...
ncn-w001:/home #
```

#### **Check to see if the images/ LUN mapping created under "targetcli ls" cmd**

> Check to see if the “targetcli ls” output shows that fileio backing store are created for rootfs images along with corresponding iSCSI LUNs

Reviewer: It's subtle to see, but the double quotes in `“targetcli ls”` are fancy multi-character sequences, please change them to simple double quotes like `"targetcli ls"` (like they are on the previous line)

Author: Fixed
